main.c

```diff
@@ -753,8 +753,16 @@ static void *tcmur_cmdproc_thread(void *arg)
 	struct tcmur_handler *rhandler = tcmu_get_runner_handler(dev);
 	struct tcmur_device *rdev = tcmu_dev_get_private(dev);
 	struct pollfd pfd;
-	int ret;
 	bool dev_stopping = false;
+	char *pname;
+	int ret;
+
```

Comment:
Make a function so we can handle errors once. Here we are just at the string len limit. We will probably never have more than 9999 LUNs (probably really 4k) so the uio name len would be at most "uio9999" (7 chars), and then for "cmdproc-uio9999" we are at 16 including the null.

We can do:

```
void tcmu_set_thread_name(const char *prefix, struct tcmu_device *dev)
{
        if ((strlen(tcmu_dev_get_uio_name(dev)) + strlen(prefix) > TCMU_THREAD_NAME_LEN - 1) ||
           ((asprintf(&pname, "cmdproc-%s", tcmu_dev_get_uio_name(dev)) == -1) {
            pthread_setname_np(pthread_self(), "cmdproc");
                pthread_setname_np(pthread_self(), "cmdproc");
       } else {
		pthread_setname_np(pthread_self(), pname);
 		free(pname);
     }
}
```

Reply:
Make sense and done.